### ISSUE TITLE
Support dark theme

### ISSUE THREAD

**Issue by peterdell** [peterdell/wudsn-ide#6]
On large monitors, working with a dark theme is mandatory to not overload the eyes. Currently the icons and color schemes in WUDSN no not work correctly when the darf theme is active. Icons need to be reworks to use transparency and probably there need to be two sets of icons and default colors.

![image](https://user-images.githubusercontent.com/10855380/211918115-3bbf22f7-f302-4ced-94aa-34ba7f1dd7da.png)

**Comment by flashjazzcat** [comment]
Sounds good. What dark theme is used in the screenshot, though? My installation has light scrollbars to the left and other horrors when I select the dark theme.
![Eclipse](https://user-images.githubusercontent.com/7622152/212060358-1d278541-a323-4e9a-9e2a-b434c0f7273f.png)

**Comment by peterdell** [comment]
From what I see, there is just one dark theme. The bit strange part is, that when you switch the first time, not all screen elements are updated and it displays a dialog where you can choose to restart. If you do that, everything looks fine. If you don't some elements retain the light skin like in your screen shot and like the window borders in the one below. It would be more reasonable if there was forced restart.

![image](https://user-images.githubusercontent.com/10855380/212184030-6f39cffc-4f1e-402e-8290-644aa1761aa4.png)

**Comment by flashjazzcat** [comment]
Yeah, I always restart after changing the theme and the result is as per my screenshot. The Outline and console scroll bars, for example, never go dark. This is on Windows 11.

**Comment by Thexder** [comment]
Might be wrong on this as I did not use Eclipse for several years. I think there is a plugin or something that makes eclipse go totally dark theme. SWT (the UI behind Eclipse) is using OS widgets. In native Windows those scrollbars are not themable.

**Comment by peterdell** [comment]
My test machine runs on Windows 10 Home N.  I'll test on a Windows 11 machine to see if that has an impact.

**Comment by peterdell** [comment]
Here's the result of my test on a current vanilla Windows 11 machine with dark mode activated system-wide. 
![image](https://user-images.githubusercontent.com/10855380/215314773-d2b88561-9fbf-4892-99c4-a7af52f71386.png)

In my eyes it is just fine, except of course for the default text colors in WUDSN. If it looks different on your machine, I suspect there might be theming settings left in your workspace. So maybe cou can try with a clean Workspace folder?
![image](https://user-images.githubusercontent.com/10855380/215314706-737e33ad-0ec3-4892-9c8e-b4476c95ed7d.png)

**Comment by flashjazzcat** [comment]
Looks good, Peter. I'll certainly give this a try. Is there a WUDSN build with the necessary transparent resource icons, if I can get the dark theme to work here?

**Comment by peterdell** [comment]
From what I see, some icons already had correct transparency (e.g. LABEL) in the past, some didn't (e.g. PROC). If you confirm there is no other major issue with the windows/scrollbars, then the icons are the next thing for me to fix.

**Comment by flashjazzcat** [comment]
Oh yes: not all icons were affected; only some. Still got the same problem with a fresh workspace, so perhaps I need to try a completely fresh install on the other PC.

![WUDSN](https://user-images.githubusercontent.com/7622152/215327727-3b5d8146-553a-4382-84ea-0679493ffe21.png)

**Comment by flashjazzcat** [comment]
OK: dark scroll bars, etc, working as you demonstrated with a fresh install of a current Eclipse version under Windows 11. Actually defaults to the OS theme on installation as well.

**Comment by peterdell** [comment]
Icon transparency added for outline etc. with 2bee22a34d1aa1d2016a3d32b5cebd23a2249597 and 05d4edd05d2a52c7dbc0a58898a3adeee2ba4a68. Mit term I should redo the icons with scale 2x.

**Comment by peterdell** [comment]
API reference https://bugs.eclipse.org/bugs/show_bug.cgi?id=546859

**Comment by peterdell** [comment]
All icons have been converted from GIF with transparency to PNG with alpha-channel. The preferences keys for the syntax highlighting colors have been separated. This way the chosen colors are kept separately for light and dark theme. The title of the group box indicates, if dark theme is active. Default colors have been added also for the dark theme. Additionally, the tooltip of the color now shows the hexadecimal color value, so you can see the exact color code.

![image](https://user-images.githubusercontent.com/10855380/221433746-8297ecc7-b29a-4cc0-b560-fe44ee7533f6.png)

**Comment by flashjazzcat** [comment]
Excellent. Will test this presently.

**Comment by peterdell** [comment]
TabFolder in preferences is still grey because of the dependency to [[Bug 566359](https://bugs.eclipse.org/bugs/show_bug.cgi?id=566359) - [Dark theme][win32]- TabFolder header not styled](https://bugs.eclipse.org/bugs/show_bug.cgi?id=566359)  which depends on [Bug 568112](https://bugs.eclipse.org/bugs/show_bug.cgi?id=568112)

**Comment by peterdell** [comment]
The problem that the default foreground text color is the preferences is not considered appears to be general Eclipse issue, too. Reported as https://github.com/eclipse-platform/eclipse.platform.ui/issues/869

**Comment by peterdell** [comment]
Note: Content assist background is also not yet correct
![image](https://user-images.githubusercontent.com/10855380/229362876-61531e05-3dba-4ccc-a9c0-65da4bb7d2aa.png)
